/**
 * Licensed to the Apache Software Foundation (ASF) under one
 * or more contributor license agreements.  See the NOTICE file
 * distributed with this work for additional information
 * regarding copyright ownership.  The ASF licenses this file
 * to you under the Apache License, Version 2.0 (the
 * "License"); you may not use this file except in compliance
 * with the License.  You may obtain a copy of the License at
 * <p>
 * http://www.apache.org/licenses/LICENSE-2.0
 * <p>
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

syntax = "proto3";

package arrow.flight.perf;

message Perf {
  bytes schema = 1;
  int32 stream_count = 2;
  int64 records_per_stream = 3;
  int32 records_per_batch = 4;
}

/*
 * Payload of ticket
 */
message Token {

  // definition of entire flight.
  Perf definition = 1;

  // inclusive start
  int64 start = 2;

  // exclusive end
  int64 end = 3;

}